apps/files/src/components/FileList.vue

```diff
@@ -301,6 +308,32 @@ export default {
       }
 
       return `files-file-list-action-button-${id}`
+    },
+
+    $_resizeHeader () {
+      this.$nextTick(() => {
+        const headerRow = this.$refs.headerRow
+        const firstRow = this.$refs.firstRow
+        if (headerRow && firstRow) {
+          const headerColumns = Array.from(headerRow.$el.childNodes).filter(n => n.nodeType === Node.ELEMENT_NODE)
+          const rowColumns = Array.from(firstRow.$el.childNodes).filter(n => n.nodeType === Node.ELEMENT_NODE)
+          for (let i = 0; i < headerColumns.length; i++) {
+            const headerColumn = headerColumns[i]
+            const rowColumn = rowColumns[i]
+            if (!rowColumn) {
+              // can happen in case of discrepancy between number of header and row columns
+              console.warn('no row column found for header ', headerColumn)
+              break
+            }
+            const rowComputedStyle = getComputedStyle(rowColumn)
+            headerColumn.classList.remove('uk-width-expand')
```

Comment:
Instead of removing ui lib specific classes, you could set both the width and max-width, to make sure that header columns can't expand to larger widths than the body columns.